build-profiling-ffi.sh

```diff
@@ -154,9 +154,6 @@ cbindgen --crate ddcommon-ffi \
 cbindgen --crate "${datadog_profiling_ffi}" \
     --config profiling-ffi/cbindgen.toml \
     --output "$destdir/include/datadog/profiling.h"
-cbindgen --crate ddtelemetry-ffi \
-    --config ddtelemetry-ffi/cbindgen.toml \
-    --output "$destdir/include/datadog/telemetry.h"
-./target/debug/dedup_headers "$destdir/include/datadog/common.h" "$destdir/include/datadog/telemetry.h" "$destdir/include/datadog/profiling.h"
+./target/debug/dedup_headers "$destdir/include/datadog/common.h" "$destdir/include/datadog/profiling.h"
```

Comment:
We need to generate the headers here for telemetry if we want to use both `datadog/profiling.h` and `datadog/telemetry.h` together, because otherwise the 2 build scripts are going to generate different `datadog/common.h` headers.
Maybe the header generation can be moved to a seprate scripts that we execute once per build.

Reply:
:thinking: 
Problem is the generated FFI library only has symbols for either profiling or telemetry.
So having extra header information might not be useful

Reply:
`common.h` only contains function declarations which are contained in both `profiling.h` and `telemetry.h` (things like ffi::Vec C bindings for instance). So generating from both headers will not expose symbols in common.h that are not in built artefacts.

But if you don't use both headers to generate it, `profiling.h` and `telemetry.h` might contain the same definitions causing collisions if you include both headers.

Reply:
Right now, I don't think you could load both telemetry and profiling dynlibs anyway because we export the same symbols from the two libs, but it's not a reason to not make the headers compatible.

Reply:
I'm curious, do we expect there to be many situations where we only make use of profiling, but not telemetry symbols, or vice-versa?

Reply:
@ivoanjo at the moment its TBD - I think. 
For PHP case separate header sets will work without a problem. 
For other Languages in the future probably a single header set will be the way to go

Reply:
Well, I'm currently anyway re-exporting everything via `pub use ddsomething::*;` to only have a single static library. I just need all my headers cooperating. (because cbindgen does not (yet) support recursive header generation across multiple crates at once)

Reply:
I'd be happy to target a single library for Ruby as well until we find good reasons to do otherwise :)